instrumentation/log4j/log4j-appender-1.2/javaagent/src/test/java/io/opentelemetry/instrumentation/log4j/appender/v1_2/Log4j1Test.java

```diff
@@ -67,6 +75,25 @@ private static Stream<Arguments> provideParameters() {
         Arguments.of(true, true));
   }
 
+  @Test
+  public void testCodeAttributes() {
+    logger.info("this is test message");
+    testing.waitAndAssertLogRecords(
+        logRecord ->
+            logRecord
+                .hasBody("this is test message")
+                .hasInstrumentationScope(InstrumentationScopeInfo.builder("abc").build())
+                .hasSeverity(Severity.INFO)
+                .hasSeverityText("INFO")
+                .hasAttributesSatisfyingExactly(
+                    equalTo(THREAD_NAME, Thread.currentThread().getName()),
+                    equalTo(THREAD_ID, Thread.currentThread().getId()),
+                    equalTo(CODE_NAMESPACE, Log4j1Test.class.getName()),
+                    equalTo(CODE_FUNCTION, "testCodeAttributes"),
+                    satisfies(CODE_LINENO, AbstractLongAssert::isPositive),
+                    equalTo(CODE_FILEPATH, "Log4j1Test.java")));
```

Comment:
you can use `io.opentelemetry.instrumentation.testing.junit.code.SemconvCodeStabilityUtil` for assertions, that was added in #13860 and allows to remove the `@SuppressWarnings` annotation by providing assertions for all the code attributes. This also allows to test both old and new conventions, but here only the old ones are tested which is fine.